src/translations/i18n.js

```diff
@@ -6,35 +6,42 @@ import {
   enUSIndividualScoreReport,
   esTranslations,
   esCOTranslations,
+  deTranslations,
   esIndividualScoreReport,
   esCOIndividualScoreReport,
 } from './exports';
+import { isLevante } from '@/helpers';
 
 export const languageOptions = {
   'en-US': { translations: enUSTranslations, language: 'English (United States)', code: 'usa' },
   en: { translations: enTranslations, language: 'English (United Kingdom)', code: 'gb' },
   es: { translations: esTranslations, language: 'Español (Spain)', code: 'es' },
   'es-CO': { translations: esCOTranslations, language: 'Español (América Latina)', code: 'col' },
+  de: { translations: deTranslations, language: 'Deutsch', code: 'de' },
 };
-export let browserLocale = window.navigator.language;
 
-const getLocale = (locale) => {
-  if (Object.keys(languageOptions).includes(locale)) {
-    console.log('Locale found in languageOptions: ', locale);
-    return locale;
-  } else if (locale.includes('es')) {
-    console.log('Spanish dialect not supported, using default es.');
-    return 'es';
+const browserLocale = window.navigator.language;
+
+const getLocale = (localeFromBrowser) => {
+  const localeFromStorage = sessionStorage.getItem(`${isLevante ? 'levante' : 'roar'}PlatformLocale`);
+
+  if (localeFromStorage) {
+    return localeFromStorage;
   } else {
-    console.log('Language not supported, using default en-US.');
-    return 'en-US';
+    sessionStorage.setItem(`${isLevante ? 'levante' : 'roar'}PlatformLocale`, localeFromBrowser);
+    return localeFromBrowser;
   }
 };
 
-const getFallbackLocale = (locale) => {
-  if (locale.includes('es')) {
+const getFallbackLocale = () => {
+  const localeFromStorage = sessionStorage.getItem(`${isLevante ? 'levante' : 'roar'}PlatformLocale`);
+
+  if (localeFromStorage.includes('es')) {
     console.log('Setting fallback local to es');
     return 'es';
+  } else if (localeFromStorage.includes('de')) {
+    console.log('Setting fallback local to de');
+    return 'de';
   } else {
```

Comment:
Why only cater for `ES` and `DE` and instead set the locale to the language code if it's defined in `languageOptions`?